sqlglot/executor/__init__.py

```diff
@@ -4,15 +4,15 @@
 from sqlglot import parse_one
 from sqlglot.executor.python import PythonExecutor
 from sqlglot.optimizer import RULES, optimize
-from sqlglot.optimizer.merge_derived_tables import merge_derived_tables
+from sqlglot.optimizer.merge_subqueries import merge_subqueries
 from sqlglot.planner import Plan
 
 logger = logging.getLogger("sqlglot")
 
 OPTIMIZER_RULES = list(RULES)
 
 # The executor needs isolated table selects
-OPTIMIZER_RULES.remove(merge_derived_tables)
+OPTIMIZER_RULES.remove(merge_subqueries)
```

Comment:
the executor i think may want some of merge subqueries, so it wants to remove some duplicate stuff but not de isolate subqueries? do you think that's feasible?

Reply:
eliminate_subqueries removes duplicates.

Reply:
Maybe we should run "isolate_table_selects" last, during the executor?

Reply:
yea that makes sense

Reply:
why do you need isolate_table_selects again? i removed it and it actually made the optimized sql look better, (predicate pushdown was further)

Reply:
i guess it hurt some of the deduplication

Reply:
ah hah! your merge_where hurts predicate pushdown, it's merging to the where when it should be merging to the join ON

Reply:
you have a bug here

```
    if isinstance(from_or_join, exp.Join) and from_or_join.side:
        # Merge predicates from an outer join to the ON clause
        from_or_join.on(where.this, copy=False)
        from_or_join.set("on", simplify(from_or_join.args.get("on")))
```

it should be not from_or_join.side since you cannot combine predicates in a left join, only inner joins

Reply:
maybe i'm wrong, maybe you can push up both wheres with parents of left join and join.

you definitely cannot push a left join into there where, because that changes left join into a filter which is wrong.

the fix may just be to do

```
if isinstance(from_or_join, exp.Join):
  ...
```